packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -351,15 +351,34 @@ const getSectionPlaceholder = <TDate>(
   }
 };
 
+const getEscapedPartsFromExpandedFormat = <TDate>(
+  utils: MuiPickersAdapter<TDate>,
+  expandedFormat: string,
+) => {
+  const escapedParts: { start: number; end: number }[] = [];
+  const { start: startChar, end: endChar } = utils.escapedCharacters;
+  const regExp = new RegExp(`(\\${startChar}[^\\${endChar}]*\\${startChar})+`, 'g');
```

Comment:
tests are passing, but the behavior does not work. I get the following message with this simple example:

```jsx
<DateField format="MM [myLitleToken<3] YYYY" />
```

> Error: MUI: The token "myLitleToken<3" is not supported by the Date and Time Pickers.

Here is the fix
```suggestion
  const regExp = new RegExp(`(\\${startChar}[^\\${endChar}]*\\${endChar})+`, 'g');
```

The reason why tests are passing is that tests run with an adapter using `'` as escaping characters, so open/close characters are the same

Reply:
Thanks for the catch ! :see_no_evil: